### PR TITLE
Commander X16 640x480 tgi loadable graphics driver

### PR DESCRIPTION
**Pull request by xlar54**
This is based on some work Greg K did for the C128's VDC, as well as the standard 320x240 code.  Having some issue with the DoCircles demo (doesnt seem to respond to kbhit() for some reason).  Ill continue to investigate, or take your comments

### CODE REVIEW

**Comment by xlar54** [comment]
image

**Comment by xlar54** [comment]
image

**Comment by xlar54** [comment]
image
image

File: libsrc/cx16/tgi/cx640p1.s [hunk 1]
```diff
@@ -0,0 +1,677 @@
...
+CLEAR:
+    .scope inner
+
+    ; set up DCSEL=2
+    lda #(2 << 1)
+    sta VERA::CTRL
+
+    ; set cache writes
+    lda #$40
+    tsb $9F29 ;VERA_FX_CTRL
```
Comment (mrdudz) [inline libsrc/cx16/tgi/cx640p1.s:256]: Why use a literal address here and not the respective constant?

File: libsrc/cx16/tgi/cx640p1.s [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+
+        .include        "zeropage.inc"
+
+        .include        "tgi-kernel.inc"
+        .include        "tgi-error.inc"
+
+        .include        "cbm_kernal.inc"
+        .include        "cx16.inc"
+
+        .macpack        generic
```
Comment (mrdudz) [inline libsrc/cx16/tgi/cx640p1.s:17]: Is this actually needed?

File: libsrc/cx16/tgi/cx640p1.s [hunk 3]
```diff
@@ -0,0 +1,677 @@
...
+        rts
+
+; ------------------------------------------------------------------------
+; CLEAR: Clear the screen.
+;
+; Must set an error code: NO
+
+CLEAR:
+    .scope inner
+
```
Comment (mrdudz) [inline libsrc/cx16/tgi/cx640p1.s:249]: Please use the same indention level for code in the entire file

File: libsrc/cx16/tgi/cx640p1.s [hunk 4]
```diff
@@ -0,0 +1,677 @@
...
+        beq @ahead
+
+        ; if BITMASK = $00, white is line color
+        ; Set the bit in the byte at VERA_DATA0
+        lda VERA::DATA0      ; Load the byte at memory address
+        ora bitMasks1,X      ; OR with the bit mask
+        sta VERA::DATA0      ; Store back the modified byte
+        rts
+
+    @ahead:
```
Comment (mrdudz) [inline libsrc/cx16/tgi/cx640p1.s:450]: Please make all labels start at the beginning of a line

Comment (xlar54) [reply libsrc/cx16/tgi/cx640p1.s]: Got it, thank you

File: libsrc/cx16/tgi/cx640p1.s (hunk 3 above)
Comment (xlar54) [reply libsrc/cx16/tgi/cx640p1.s:249]: Got it!

File: libsrc/cx16/tgi/cx640p1.s (hunk 2 above)
Comment (xlar54) [reply libsrc/cx16/tgi/cx640p1.s:17]: Apparently so.  I believe (not sure really) that module_header requires this.  Removing the ,macpack lines caused errors in compilation that admittedly I dont fully understand.  But it the macpack lines exist in the Commodore tgi drivers as well.

File: libsrc/cx16/tgi/cx640p1.s (hunk 1 above)
Comment (xlar54) [reply libsrc/cx16/tgi/cx640p1.s]: replaced in latest, thank you

**Comment by mrdudz** [comment]
OK, merging... thanks!

**Pull request merged**